libs/ui/src/unlock_machine_screen.tsx

```diff
@@ -55,6 +59,10 @@ export function UnlockMachineScreen({ auth }: Props): JSX.Element {
   }, []);
 
   useEffect(() => {
+    if (isFeatureFlagEnabled(EnvironmentFlagName.SKIP_PIN_ENTRY)) {
+      auth.checkPasscode(auth.user.passcode);
```

Comment:
Should we `return` after this?

Reply:
The rest of the effect won't run because there is no security pin to check, so it shouldn't matter, but I think it's a good idea in case there's edge cases I can't think of.